### PR TITLE
feat: login and register done

### PR DESCRIPTION
**Pull request by felixdusengimana**
#### What does this PR do?
 
Implementing login and register on frontend.

#### Description of  the task to be completed?

I have done implementation of login and register on frontend part with user

#### How should this be manually tested?

Check out this repo and then run `npm i` after successful installation run `npm run dev` then go into browser at this URL `{BASE_URL}/auth/login` or `{BASE}/auth/register` for register.

#### Any background context you want to provide?

I have used `react-query` so someone implementing other request via frontend should understand it.

#### Screenshots (if appropriate)

image

image

### CODE REVIEW

File: src/plugins/axios.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type {AxiosRequestConfig, InternalAxiosRequestConfig} from 'axios';
+import axios from 'axios';
+import { LoginRes } from '../types/services/auth.types';
+import cookies from '../utils/cookies';
+
+export const BASE_URL = 'http://localhost:5000/api'
+const config: AxiosRequestConfig = {
+	baseURL: BASE_URL,
+};
```
Comment (Nkbtemmy) [inline src/plugins/axios.ts]: Please use .env file for accessing BASE_URL as well as other credentials

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]
Remember to check UI/UX before pushing !!

**Comment by felixdusengimana** [comment]
> Remember to check UI/UX before pushing !!

> Let me solve that

**Review by Nkbtemmy — APPROVED** [review]

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]
You need to rebase updated develop

**Comment by Nkbtemmy** [comment]
Or do npm install before you push

**Review by Nkbtemmy — APPROVED** [review]

**Pull request merged**